caluma/form/serializers.py

```diff
@@ -324,3 +333,82 @@ class SaveDocumentFloatAnswerSerializer(SaveAnswerSerializer):
 
     class Meta(SaveAnswerSerializer.Meta):
         pass
+
+
+class SaveDocumentTableAnswerSerializer(SaveAnswerSerializer):
+    value = serializers.GlobalIDPrimaryKeyRelatedField(
+        source="documents",
+        queryset=models.Document.objects,
+        many=True,
+        required=True,
+        help_text="List of document ids whereas one document is one row in the table",
+    )
+
+    def validate(self, data):
+        documents = (
+            data.get("documents") or self.instance and self.instance.documents or []
+        )
+        question = data.get("question") or self.instance and self.instance.question
+
+        for document in documents:
+            if document.form_id != question.row_form_id:
+                raise exceptions.ValidationError(
+                    f"Document {document.pk} is not of form type {question.form.pk}."
+                )
+
+        return super().validate(data)
+
+    def create_answer_documents(self, answer, documents):
+        family = answer.document.family
+        document_ids = [document.pk for document in documents]
+
+        for sort, document_id in enumerate(reversed(document_ids)):
+            models.AnswerDocument.objects.create(
+                answer=answer, document_id=document_id, sort=sort
+            )
+
+        models.Document.objects.filter(
+            family__in=models.Document.objects.filter(pk__in=document_ids).values(
+                "family"
+            )
+        ).update(family=family)
+
+    @transaction.atomic
+    def create(self, validated_data):
+        documents = validated_data.pop("documents")
+        instance = super().create(validated_data)
+        self.create_answer_documents(instance, documents)
+        return instance
+
+    def _get_document_tree(self, document_id):
+        answers = models.AnswerDocument.objects.filter(document_id=document_id).values(
+            "answer"
+        )
+        child_documents = models.Document.objects.filter(answers=answers).distinct()
+
+        for child_document in child_documents:
+            yield from self._get_document_tree(child_document.pk)
+
+        yield document_id
+
+    @transaction.atomic
+    def update(self, instance, validated_data):
+        documents = validated_data.pop("documents")
+
+        # detach answers to its own family tree
```

Comment:
The comment is confusing me a bit. Aren't we detaching answers from their old family tree and afterwards reattaching them to the new one?